src/data_provider/CMakeLists.txt

```diff
@@ -14,7 +14,27 @@ if(COVERITY)
   add_definitions(-D__GNUC__=8)
 endif(COVERITY)
 
-set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wshadow -Wnon-virtual-dtor -Woverloaded-virtual -Wunused -Wcast-align -Wformat=2 -std=c++14")
+#Check if filesystem is available
+include(CheckCXXSourceCompiles)
+
+check_cxx_source_compiles("
```

Comment:
This doesn't work in my Ubuntu 20.04, with `g++ (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0` inside CMake, but it does manually if c++17 is enabled

```bash
root@ubuntu-focal:/home/vagrant/wazuh# g++ -o test test.cpp 
test.cpp: In function ‘int main()’:
test.cpp:3:14: error: ‘std::filesystem’ has not been declared
    3 |         std::filesystem::path p;
      |              ^~~~~~~~~~
root@ubuntu-focal:/home/vagrant/wazuh# g++ -o test test.cpp -std=c++17
root@ubuntu-focal:/home/vagrant/wazuh# 

```

So in that exploratory test, no `python` packages were obtained.

Reply:
good catch!